### PR TITLE
Implement ConfirmedWithdrawIntent handler

### LINKED ISSUE #295: Implement ConfirmedWithdrawIntent handler
- Upserts `WithdrawIntent` model
- Upserts the `Message` model.

### PR DESCRIPTION
**Pull request by Vindeep07**
Fixes #295 

- Implemented ConfirmedWithdrawIntent Handler
- Added tests for handle function

### CODE REVIEW

File: src/m1_facilitator/handlers/ConfirmWithdrawIntentHandler.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import MessageRepository from '../repositories/MessageRepository';
+import Message, { MessageStatus, MessageType } from '../models/Message';
+import GatewayRepository from '../repositories/GatewayRepository';
+
+import assert = require('assert');
```
Comment (0xsarvesh) [inline src/m1_facilitator/handlers/ConfirmWithdrawIntentHandler.ts]: External import should be at the top.

File: src/m1_facilitator/handlers/ConfirmWithdrawIntentHandler.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+   * - Handles the ConfirmWithdrawIntent entity.
+   * - Updates the target status of message with `Withdraw` type to `Declared`
+   *   if it is `Undeclared`.
+   * - If message does not exists, it creates message record.
+   * - This handler only reacts to the events of gateways which are populated
+   *   during seed data. It silently ignores the events by the other gateways.
+   *
+   * @param records List of ConfirmWithdrawIntents.
+   */
+  public async handle(records: {
```
Comment (0xsarvesh) [inline src/m1_facilitator/handlers/ConfirmWithdrawIntentHandler.ts]: Please Extra this inline type to an interface `ConfirmWithdrawIntentEntityInterface`

File: src/m1_facilitator/models/Gateway.ts [hunk 3]
```diff
@@ -38,9 +38,9 @@ export default class Gateway extends Comparable<Gateway> {
 
   public gatewayType: GatewayType;
 
-  public destinationGA?: string;
+  public remoteGatewayLastProvenBlockNumber: BigNumber;
```
Comment (0xsarvesh) [inline src/m1_facilitator/models/Gateway.ts]: can you rebase your PR? This change is already merged, shouldn't show in diff.

File: src/m1_facilitator/handlers/ConfirmWithdrawIntentHandler.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
+// Copyright 2020 OpenST Ltd.
```
Comment (0xsarvesh) [inline src/m1_facilitator/handlers/ConfirmWithdrawIntentHandler.ts]: Name of the file should be ConfirmWithdrawIntentsHandler.

File: test/m1_facilitator/handlers/ConfirmWithdrawIntentHandler/handle.test.ts [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+    const message = await messageRepository.get(messageHash);
+
+    assert.strictEqual(
+      message && message.targetStatus,
+      MessageStatus.Declared,
+      'Target status should be declared',
+    );
+  });
+
+  it('should create a message with target status declared'
```
Comment (0xsarvesh) [inline test/m1_facilitator/handlers/ConfirmWithdrawIntentHandler/handle.test.ts]:
```suggestion
  it('should create a message with target status declared'
```

**Review by 0xsarvesh — CHANGES_REQUESTED** [review]
Overall looks nice 👍 
Some minor comments.

File: src/m1_facilitator/handlers/ConfirmWithdrawIntentHandler.ts (hunk 1 above)
Comment (deepesh-kn) [reply src/m1_facilitator/handlers/ConfirmWithdrawIntentHandler.ts]: Checked the linter the ordering is correct.

File: src/m1_facilitator/handlers/ConfirmWithdrawIntentHandler.ts (hunk 2 above)
Comment (deepesh-kn) [reply src/m1_facilitator/handlers/ConfirmWithdrawIntentHandler.ts]: Extracted as `ConfirmWithdrawIntentsEntityInterface`

File: src/m1_facilitator/models/Gateway.ts (hunk 3 above)
Comment (deepesh-kn) [reply src/m1_facilitator/models/Gateway.ts]: Rebased.

File: src/m1_facilitator/handlers/ConfirmWithdrawIntentHandler.ts (hunk 4 above)
Comment (deepesh-kn) [reply src/m1_facilitator/handlers/ConfirmWithdrawIntentHandler.ts]: Changed to `ConfirmWithdrawIntentsHandler`

File: test/m1_facilitator/handlers/ConfirmWithdrawIntentHandler/handle.test.ts (hunk 5 above)
Comment (deepesh-kn) [reply test/m1_facilitator/handlers/ConfirmWithdrawIntentHandler/handle.test.ts]: Fixed.

**Review by 0xsarvesh — DISMISSED** [review]
LGTM 👍

**Review by 0xsarvesh — APPROVED** [review]
LGTM 🚀

**Pull request merged**